### DISCUSSION TITLE
Export to xml

### DISCUSSION THREAD

**Discussion by Oddivh** [freeplane/freeplane#1032] [Help]
Hi,
When exporting a map to XML, I get the following message: "Export Failed" and the file in empty. Anyone knows why?

**Comment by marco64io** [comment]
One reason could be that the names of nodes have characters like underscore, apostrophe etc. that are not permitted in XML file

**Reply by Oddivh** [reply]
Thank you, you are right. The export to XML does not handle space.

![image](https://user-images.githubusercontent.com/126766998/222395055-94673233-4a55-4a9d-ad8b-a08994e1890a.png)

**Comment by marco64io** [comment]
Yes, is the space that XML doesn't handle, not the underscore as I wrote

**Reply by dpolivaev** [reply]
This export is supposed to work as a reverse operation to File->Import->Import xml file. It tries to convert node text as XML element names and fails if they contain any characters which can not be there.

What is your goal?

**Reply by Oddivh** [reply]
![image](https://user-images.githubusercontent.com/126766998/223040103-3faab2de-d5a0-4c62-a707-4aa8a3e03a09.png)

**Reply by ldmpub** [reply]
Hi,

I have started to use Freeplane XML export to draft the structure of an export file from a Freeplane MindMap.
Context: in a webapp, I want to design a feature allowing user to export all his datas in a XML file.

I have noticed that I need to select a branch and use "branch export" to make Freeplane XML export work but for unknown reasons:
* when using "map export" => generated XML contains no data
* when using "branch export" with map root node selected => generated XML contains no data

 ... can someone help me understand what's going on?

Attached a sample map to reproduce by using Map and Branch XML exports => [Structure.zip](https://github.com/freeplane/freeplane/files/11524158/Structure.zip)
